### PR TITLE
SI-8052 Disallow `macro` as an identifier

### PR DESCRIPTION
**Pull request by soc**
Note that the change could look a lot cleaner, at the cost of returning
more generic error messages. I decided that at least for 2.11 I'll keep
scalac remembering that macro was a standard identifier name once, so
that we can point out more precisely what's wrong with users' code.

### CODE REVIEW

**Comment by soc** [comment]
Review by @xeno-by, please!

File: test/files/neg/macro-deprecate-idents.check [hunk 1]
```diff
@@ -1,54 +1,67 @@
...
+^
+macro-deprecate-idents.scala:42: error: '{' expected but ';' found.
+package foo {
+^
+macro-deprecate-idents.scala:45: error: '{' expected but '}' found.
+}
+^
+macro-deprecate-idents.scala:52: error: ')' expected but '}' found.
+}
+^
```
Comment (xeno-by) [inline test/files/neg/macro-deprecate-idents.check:66]: I wonder whether it's possible to avoid reporting these spurious syntactic errors.

Comment (soc) [inline test/files/neg/macro-deprecate-idents.check:66]: Me, too. I'm wondering whether this was expected or just a side-effect of the change.

Comment (xeno-by) [inline test/files/neg/macro-deprecate-idents.check:66]: For one, as a user, I wouldn't expect these errors.

Comment (soc) [inline test/files/neg/macro-deprecate-idents.check:66]: I totally agree, not sure if it matters, though considering the likely total amount of times people will see this error.

Comment (xeno-by) [inline test/files/neg/macro-deprecate-idents.check:66]: WDYT, @retronym?

Comment (retronym) [inline test/files/neg/macro-deprecate-idents.check:66]: I tried to improve it but failed. It is no worse than

  val class = 0

So no need to block this pr.

**Comment by xeno-by** [comment]
LGTM

**Pull request merged**